addon/MetadataStore.js

```diff
@@ -425,25 +425,28 @@ MetadataStore.prototype = {
   /**
    * Get page metadata (including images) for the given cache_keys.
    * For the missing cache_keys, it simply ignores them and will not
-   * raise any exception. Note that if this function gets called on
-   * a closed or unestablished connection, it returns an empty array
+   * raise any exception. Note that this function throws an exception
+   * if it is called with a closed or unestablished connection
    *
    * @param {Array} cacheKeys an cache key array
    *
    * Returns a promise with the array of retrieved metadata records
    */
   asyncGetMetadataByCacheKey: Task.async(function*(cacheKeys) {
-    const quoted = cacheKeys.map(key => `'${key}'`).join(",");
+    const quoted = cacheKeys.map(key => "?").join(",");
     let metaObjects;
     try {
       metaObjects = yield this.asyncExecuteQuery(
         `SELECT * FROM page_metadata WHERE cache_key IN (${quoted})`,
```

Comment:
I'm pretty sure you could've done just `:keys` then `params: {keys: cacheKeys}` below. Kind of like you did the other query. Did you try that?